mache/cime_machine_config/config_machines.xml

```diff
@@ -256,15 +257,14 @@
     <environment_variables>
       <env name="MPICH_ENV_DISPLAY">1</env>
       <env name="MPICH_VERSION_DISPLAY">1</env>
+      <env name="MPICH_MPIIO_DVS_MAXNODES">1</env>
       <env name="OMP_STACKSIZE">128M</env>
       <env name="OMP_PROC_BIND">spread</env>
       <env name="OMP_PLACES">threads</env>
       <env name="HDF5_USE_FILE_LOCKING">FALSE</env>
       <env name="PERL5LIB">/global/cfs/cdirs/e3sm/perl/lib/perl5-only-switch</env>
       <env name="FI_CXI_RX_MATCH_MODE">software</env>
       <env name="MPICH_COLL_SYNC">MPI_Bcast</env>
-      <env name="Albany_ROOT">$SHELL{if [ -z "$Albany_ROOT" ]; then echo /global/common/software/e3sm/mali_tpls/albany-e3sm-serial-release-gcc; else echo "$Albany_ROOT"; fi}</env>
-      <env name="Trilinos_ROOT">$SHELL{if [ -z "$Trilinos_ROOT" ]; then echo /global/common/software/e3sm/mali_tpls/trilinos-e3sm-serial-release-gcc; else echo "$Trilinos_ROOT"; fi}</env>
```

Comment:
We need to add and delete them when we can.  The Albany and Trilinos ones we didn't have before.  But the others we should have had or explicitly commented on why we didn't:
https://github.com/E3SM-Project/mache/blob/main/mache/spack/pm-cpu_gnu_mpich.sh#L42-L51
Again, it looks like there's some work to be done here and things have gotten out of date.

Reply:
Again, the environment variables apply to all compilers and MPI.  But there is different syntax for `.sh` and `.csh` so that's something to be careful about.

Reply:
For now, just put in all the environment variables, and I'll help sort out what we need or don't need.  But I can tell you now that we don't need the `Albany_ROOT` and `Trilinos_ROOT` ones.

Reply:
So, would you like me to add all missing environment variables (whether or not they were added in this update or beforehand), unless they are commented out?

Reply:
Yes, exactly.  I'll need to look carefully at the results then as part of my review (and also test) with Compass or Polaris or both before we merge this.